### ISSUE TITLE
Target HPKE draft-08

### ISSUE THREAD

**Issue by sayrer** [briansmith/ring#1209]
The TLS WG is proposing to update the interop version with a new draft that would target cfrg-hpke-draft-08, and hopefully eliminate churn there. I think it would be best to target that version in Ring. In Rustls, we can prototype against Cloudflare's deployed ECH draft-09 code (cfrg-hpke-draft-07) using an old version of the rust-hpke crate.

See: https://mailarchive.ietf.org/arch/msg/tls/KluT9tockL5kDcIaevGOhyLWUDc/

**Comment by cjpatton** [comment]
Bumping this issue, as I'd also love to see this happen.

**Comment by cjpatton** [comment]
Are the maintainers open to a PR for adding partial support for the latest draft (HPKE-12)? An implementation of base mode and support for at least one cipher suite would be useful.

**Comment by briansmith** [comment]
> Are the maintainers open to a PR for adding partial support for the latest draft (HPKE-12)? An implementation of base mode and support for at least one cipher suite would be useful.

Yes, that would be great as I seem to never find time to get to this. Happy to meet with you to discuss it before you start, if that would help.

**Comment by cjpatton** [comment]
Thanks @briansmith, I'll ping you when I have some time to start work on this. Is brian@briansmith.org a good email address (from your GH profile)

**Comment by briansmith** [comment]
> Thanks @briansmith, I'll ping you when I have some time to start work on this. Is [brian@briansmith.org](mailto:brian@briansmith.org) a good email address (from your GH profile)

Great! Yes, that's the email address.

**Comment by tanyav2** [comment]
PR implementing base mode is now up for initial review- https://github.com/briansmith/ring/pull/1462
